### ISSUE TITLE
Calling R code from within Spark(Scala)

### ISSUE THREAD

**Issue by gaganverma1** [dbdahl/rscala#15]
I am trying to use R code/libraries in a Spark app written in scala using rscala. I am getting the following error
 
`Exception in thread "main" java.lang.NoClassDefFoundError: org/ddahl/rscala/RClient$`

I am building the app using SBT and runnig on the local machine using `spark-submit.` Also I have aready updated the `build.sbt` file with Spark and rscala dependencies:

`libraryDependencies += "org.apache.spark" %% "spark-core" % "2.4.2"`
`libraryDependencies += "org.ddahl" %% "rscala" % "3.2.6"`
 I tried older version of `rscala 2.5.0` but got the same error. 

Please let me know if I am missing something. Is it possible to use rscala with Spark?

**Comment by dbdahl** [comment]
I would first suggest making sure you can successfully use rscala in a simple, standard Scala application.  I am aware of Spark, but have not used it.  I am happy to investigate if you could provide a simple Spark example (and usage instructions) that reproduces the issue.  You can reach me via email at dahl@stat.byu.edu.

**Comment by gaganverma1** [comment]
Thanks a lot for your response. I tried to solve this over the weekend and figured out that I need to bundle the `rscala` library (which is considered as an external dependency)  with standard Spark libraries, and make a assembly/FAT jar file using SBT's `assembly` plugin. Just mentioning rscala dependencies in `build.sbt` was not sufficient as it was not being picked up at run time and hence I was getting the error. I think this might be the case with any other external library in Spark and not just with `rscala`. When I `spark-submit` the assembly jar I was able to run my Spark app. 
If I just run a simple Scala app (without Spark) as you suggested then assemble jar was not required and the app was runnig fine just with normal `build.sbt `
